### ISSUE TITLE
Investigate Source Tree View

### ISSUE THREAD

**Issue by clarkbw** [jlongster/debugger.html#104]
**User Story**

As a developer I would like the various JS source files organized by domain and folder such that I can easily navigate the hierarchy to find the files I'm looking for.

- [ ] Tree Node for each non-empty directory

**Comment by markerikson** [comment]
Please.  This is one of the core reasons why I currently use Chrome DevTools as my primary debugging environment.  (Also the fact that the FF DevTools don't seem to have a single "Breakpoints" pane like Firebug used to, but rather breakpoints are hidden by what file they're in.)

**Comment by markerikson** [comment]
Also: there's several React-based treeviews out there.  I spent a chunk of time cataloging them for my own project at work, and have a bunch of them bookmarked at https://pinboard.in/search/u:acemarke?query=widget%2Btree .  However, I actually wound up writing my own.  The current implementation is specific to my app, but if there was interest I might be able to extract a more generic version and throw up a sample project.

**Comment by markerikson** [comment]
Hmm.  Actually, my comments maybe shoulda gone on #103.

**Comment by clarkbw** [comment]
> Hmm. Actually, my comments maybe shoulda gone on #103.

Yes, we should have linked those two together previously.

**Comment by clarkbw** [comment]
Also, thanks for you comments @markerikson 

We have a unified break point list now, take a look at a recent screenshot:
![screen shot 2016-05-03 at 20 21 10](https://cloud.githubusercontent.com/assets/2134/15004505/bf8011a8-116c-11e6-9adf-b2df451134bd.png)

**Comment by jlongster** [comment]
Been working on this most of today. I should have something based on Nick's tree pretty soon, and Greg has implemented a basic tree with Honza's. I'll coordinate with him to get this resolved.

<img width="291" alt="screen shot 2016-05-10 at 5 16 56 pm" src="https://cloud.githubusercontent.com/assets/17031/15162883/5c2f41de-16d3-11e6-8ea7-b33d01f87369.png">

**Comment by jasonLaster** [comment]
Yay!

**Comment by TatumCreative** [comment]
I have the code in a [branch](https://github.com/TatumCreative/debugger.html/tree/tree-view-honza) with honza's component over here: https://github.com/TatumCreative/debugger.html/commit/12e5762437159584cbd9367889002f7a4ca3b8df

**Comment by jlongster** [comment]
Greg and I talked and while both trees could be sufficient, the one the memory tool uses is much simpler and lower-level and easier to leverage. That also means it is much simpler to use in our separate project (just copy the 1 file). While it needs some love to be more user-friendly, once I figured out how to use the low-level API it was easy to wrap a SourcesTree around it.

I'll be working with others on the team to standardize trees, but for now I think moving forward with the memory tool one is good. If we want to show a good use case for using npm modules, we can show how easy it is to use [react-virtualized](https://bvaughn.github.io/react-virtualized/) to integrate virtualization with this tree component. I've already done this, because the memory tool had previously embedded virtualization in their tree but I removed it because we don't need it (for now at least).

I should be open up a PR tomorrow. Working out how best to handle the data in the reducers and when to compile the sources data into nested data. But the tree is working well, and the memory tool's tree already has stuff like keyboard navigation which is great.
